### PR TITLE
Refactor cider-find-reusable-repl-buffer

### PR DESCRIPTION
**Pull request by expez**
- Remove unneeded liveness test for processes.
- Extract local function to toplevel function.
- Rewrite to favor if-let.

Related to #1385.

Could use a few more eyes on this.

### CODE REVIEW

File: cider.el [hunk 1]
```diff
@@ -217,6 +217,18 @@ should be the regular Clojure REPL started by the server process filter."
              "code" cider-cljs-repl)
        (cider-repl-handler (current-buffer))))))
 
+(defun cider--select-zombie-buffer (repl-buffers)
+  "Return a zombie buffer from REPL-BUFFERS, or nil if none exists."
+  (when-let ((zombie-buffs (seq-remove
+                            (lambda (buff) (get-buffer-process buff))
```
Comment (Malabarba) [inline cider.el]: a.k.a., `#'get-buffer-process` :-)

File: cider.el [hunk 2]
```diff
@@ -225,39 +237,23 @@ and has no process, return it.  If the process is alive, ask the user for
...
+  (if-let ((repl-buffers (cider-repl-buffers))
+           (exact-buff (seq-find
+                        (lambda (buff)
+                          (with-current-buffer buff
+                            (or (and endpoint
+                                     (equal endpoint nrepl-endpoint))
+                                (and project-directory
+                                     (equal project-directory nrepl-project-dir)))))
+                        repl-buffers)))
+      (if-let ((process (get-buffer-process exact-buff)))
```
Comment (Malabarba) [inline cider.el]: Not sure if I'm reading this right, but looks like `process` is not used here, so it could just an `if`.

**Comment by Malabarba** [comment]
Much more readable :+1:

**Comment by bbatsov** [comment]
:+1: Looks good to me.

**Pull request merged**